azext_iot/digitaltwins/providers/twin.py

```diff
@@ -214,6 +230,36 @@ def delete_relationship(self, twin_id, relationship_id, etag=None):
         except ErrorResponseException as e:
             raise CLIError(unpack_msrest_error(e))
 
+    def delete_all_relationship(self, twin_id, etag=None):
+        relationships = self.list_relationships(twin_id, incoming_relationships=True)
+        incoming_pager = self.list_relationships(twin_id)
+
+        # relationships pager needs to be advanced to get relationships
+        try:
+            while True:
+                relationships.extend(incoming_pager.advance_page())
+        except StopIteration:
+            pass
+
+        print(f"Found {len(relationships)} relationship(s) associated with twin {twin_id}.")
+
+        for relationship in relationships:
+            try:
+                if type(relationship) == dict:
+                    self.delete_relationship(
+                        twin_id=twin_id,
+                        relationship_id=relationship['$relationshipId'],
+                        etag=etag
+                    )
+                else:
```

Comment:
We should look into optimizing `list_relationships` to always return the same type so we don't need to branch here. End user usage I believe its always a list of dictionaries/list of objects returned.

Reply:
This issue stems from incoming_relationships=True being unpacked by the sdk (and giving a list of twins) vs incoming_relationships=False giving a pager and forcing us to unpack.

I also use is this a dict or twin object as a lazy way to differentiate if I should be using the twin_id or source_id when deleting the relationship. Otherwise I would have two for loops with with almost identical code structure for two lists.